services/agents/registry.go

```diff
@@ -408,6 +414,33 @@ func (r *Registry) SendSetStateRequest(ctx context.Context, pmmAgentID string) {
 	l.Infof("SetState response: %+v.", res)
 }
 
+func (r *Registry) sendAgentMetadata(stream agentpb.Agent_ConnectServer, agentID string) error {
+	ctx := stream.Context()
+	l := logger.Get(ctx)
+
+	// TODO: Refactor. Use repository or service to get an agent.
+	row := &models.Agent{AgentID: agentID}
+	if err := r.db.Reload(row); err != nil {
+		if err == reform.ErrNoRows {
+			return status.Errorf(codes.NotFound, "No Agent with ID %q.", agentID)
+		}
+		return errors.Wrap(err, "failed to find agent")
+	}
+	if row.AgentType != models.PMMAgentType {
+		return status.Errorf(codes.NotFound, "No pmm-agent with ID %q.", agentID)
+	}
+
+	header := metadata.Pairs(
+		"pmm-agent-node-id", pointer.GetString(row.RunsOnNodeID),
+		"pmm-managed-version", version.Version,
+	)
+	l.Infof("Sending metadata: %s", header)
+	if err := stream.SendHeader(header); err != nil {
+		return status.Errorf(codes.Internal, "Can't send server metadata to client.")
+	}
+	return nil
+}
```

Comment:
Let's move that to `register` method. Once it returns, agent is either fully registered and connected, or not.